packages/react-core/src/components/Menu/examples/Menu.md

```diff
@@ -18,74 +18,111 @@ import AngleLeftIcon from '@patternfly/react-icons/dist/esm/icons/angle-left-ico
 ```
 
 ### With links
 
+Use the `to` property to add a link to a `<MenuItem>` that directs users to a new page when the item is selected. Use the `isExternalLink` property when linking to external resources. This will annotate a menu item link with an external link icon when they navigate to the link or hover over it.
+
 ```ts file="MenuWithLinks.tsx"
 ```
 
-### With separator(s)
+### With descriptions
 
-```ts file="MenuWithSeparators.tsx"
+Use the `description` property to add short descriptive text below any menu item that needs additional context. 
+
+```ts file="MenuWithDescription.tsx"
 ```
 
-### With titled groups
+### Item checkbox
 
-```ts file="MenuWithTitledGroups.tsx"
+Use the `hasCheck` property to add a checkbox to a `<MenuItem>`. Use the `isSelected` property to indicate when a `<MenuItem>` is selected.
+
+```ts file="./MenuWithCheckbox.tsx" isBeta
 ```
 
-### With description
+### Menu footer
 
-```ts file="MenuWithDescription.tsx"
+Add a `<MenuFooter>` that contains separate, but related actions at the bottom of a menu.
+
+```ts file="MenuWithFooter.tsx"
 ```
 
-### With actions
+### Separated items
 
-```ts file="MenuWithActions.tsx"
+Use a (divider)[/components/divider] to visually separate `<MenuContent>`. Use the `component` property to specify the type of divider component to use. 
+
+```ts file="MenuWithSeparators.tsx"
+```
+
+### Titled groups of items
+
+Add a `<MenuGroup>` to organize `<MenuContent>` and use the `label` property to title a group of menu items. Use the `labelHeadingLevel` property to assign a heading level to the menu group. 
+
+```ts file="MenuWithTitledGroups.tsx"
 ```
 
 ### With favorites
 
+The following menu example allows users to favorite menu items, an action that duplicates a menu item and places it in a separated group at the top of the menu. The `isFavorited` property identifies items that have been favorited by a user.
+
 ```ts file="MenuWithFavorites.tsx"
 ```
 
+### Filtering with text input
+
+A (text input)[components/text-input] component can be loaded within `<MenuInput>` to load a search input at the top of the menu that filters menu items as a user types.
+
+```ts file="MenuFilteringWithTextInput.tsx"
+```
+
 ### Option single select
 
+The following example demonstrates an option select menu that persists a selected menu item. Use the `selected` property to label a selected item with a checkmark. A `<MenuItem>` may use the `isSelected` property to indicate that it is selected.
+
 ```ts file="MenuOptionSingleSelect.tsx"
 ```
 
 ### Option multi select
 
+You may need to persist multiple selections that a user makes, as demonstrated below.
+
 ```ts file="MenuOptionMultiSelect.tsx"
 ```
 
 ### With drilldown
 
+Use a drilldown menu to contain different levels of menu items. When a parent menu item (an item that has a submenu of children) is selected, the menu is replaced with the children items. A header displays the name of the parent, with the option to go back one level.
```

Comment:
@kmcfaul would you have a suggestion for what else could be included for the drilldown examples (or just this initial one)? I feel like at least it'd be worth mentioning the `<DrilldownMenu>` component.

Reply:
I changed up the drilldown examples (can't tell what's going on with the submenu one because of the visual issues). These may have been good changes, or they may have been bad -- lmk! I was trying to refer more generally to the related properties than the specific example implementation variables, but I don't feel confident that this is more effective

Reply:
Adding some of those properties I think definitely helps!

In terms of the Initially drilled-in menu example, I'm wondering if the verbiage regarding `onGetMenuHeight` should be mentioned in the first drilldown example instead, since it's a property used in all of the drilldown examples.

Regarding the submenu functions example, I believe this might be the PR where it was first introduced: https://github.com/patternfly/patternfly-react/pull/6872